docs/TuxMutableClass.md

```diff
@@ -0,0 +1,142 @@
+
+###### <a id="mutableTraits-state"></a>Property - 'mutableTraits.state' - type: STRING, ARRAY of STRINGs, ARRAY of ARRAYs of STRINGs - optional [#](#mutableTraits-state)
+A trait, or array of traits, or an array of arrays of traits, within the components state object to create a path for.
+
+```javascript
+    var mutableClassProps = {};
+    mutableClassProps.mutableTraits = {
+      props: 'text',
+      state: 'editing'
+    };
+
+    // OR
+    mutableClassProps.mutableTraits = {
+      props: ['text', 'timestamp'],
+      state: 'editing'
+    };
+
+    // OR it is possible to specify the hierarchy of keys to search through to improve performance of the one time deep search/provide greater specificity regarding the required property
+    mutableClassProps.mutableTraits = {
+      props: [
+        ['message', 'text'],
+        ['message', 'timestamp']
+      ],
+      state: 'editing'
+    };
+```
+
+##### <a id="mutableClassProps-render"></a>Parameter - `mutableClassProps.render` - type: FUNCTION - required [#](#createMutableClass-render)
+Since this is an implementation of React.createClass a render method is required.
+
+#### <a id="createMutableClass-mutableClass"></a>Return - 'mutableClass' - type: OBJECT [#](#createMutableClass-mutableClass)
+Returns a React class augmented with special Tux mixins.
+
+##### <a id="mutableClass-componentWillMount"></a>Method - 'mutableClass.componentWillMount' - type: FUNCTION [#](#mutableClass-componentWillMount)
+Component life cycle event that creates an array of string key paths attached to the component's constructor with a key of __tuxMutableTraits__. These paths are used in the shouldComponentUpdate life cycle event to compare current and next values of the specified traits to determine if the component should re-render. React automatically invokes this method immediately prior to the component mounting.
+
+##### <a id="mutableClass-shouldComponentUpdate"></a>Method - 'mutableClass.shouldComponentUpdate' - type: FUNCTION [#](#mutableClass-shouldComponentUpdate)
+Component life cycle event determines whether or not the component should update, by checking to see if any of the traits in the array of trait paths defined in the componentWillMount life cycle have changed. React automatically provides the `nextProps` and `nextState` inputs.
+
+## <a id="TuxMutableClass-Guidelines"></a>TuxMutableClass Guidelines [#](#TuxMutableClass-Guidelines)
+It is possible to use `Tux` without ever using the mutable class, yet we think you would be missing out by not using this component generator. The mutable class should not be used everywhere and there are cases where it will beneficial to use it versus not.
+
+We recommend using the mutable class in situations where you have multiples of the same type of component, where they possess deeper properties on state or props that will change occasionally or even more frequently.
```

Comment:
Change to ...in situations where you are rendering many components of the same class, and only some of the state and props in those components will ever update.